### PR TITLE
feat(ui5-flexible-column-layout): new property accessibilityRoles

### PR DESCRIPTION
**Pull request by dobrinyonkov**
Summary:

1. New property accessibilityRoles added to set custom roles on the containers (start & mid & end columns + arrow containers);
2. Add 2 additional properties to the accessibilityTexts object to set custom labels on the arrow containers);
3. Hidden text elements are placed inside the columns to ensure that all elements are contained by landmarks;

Fixes: #3804, #3796

### CODE REVIEW

**Review by nnaydenow — APPROVED** [review]
Code seems to be clear. Let's wait @xtatica and @elenastoyanovaa to approve documentation changes.

**Review by elenastoyanovaa — APPROVED** [review]

File: docs/2-advanced/07-accessibility.md [hunk 1]
```diff
@@ -55,18 +55,19 @@ For more information regarding the available themes and how to use them, see the
...
-| `interactive`                               | `tabindex`        | Defines if the component is interactive (focusable and pressable).                                                                                                                |
+| UI5 Web Components Property                 | HTML attribute  | Description                                                                                                                                                                                         |
+| ------------------------------------------- | --------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `accessibleName`                            | `aria-label`    | Defines the text alternative of the component. If not provided a default text alternative will be set, if present.                                                                                  |
```
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:60]:
```suggestion
| `accessibleName`                            | `aria-label`    | Defines the text alternative of the component. If not provided, a default text alternative is set, if present.                                                                                  |
```

File: docs/2-advanced/07-accessibility.md [hunk 2]
```diff
@@ -55,18 +55,19 @@ For more information regarding the available themes and how to use them, see the
...
+| `accessibleNameRef`                         | `aria-label`    | Alternative for `aria-labelledby`. Receives id (or many ids) of the elements that serve as labels of the component. Those labels are passed as a concatenated string to the `aria-label` attribute. |
```
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:61]:
```suggestion
| `accessibleNameRef`                         | `aria-label`    | Alternative for `aria-labelledby`. Receives ID (or many IDs) of the elements that serve as labels of the component. Those labels are passed as a concatenated string to the `aria-label` attribute. |
```

File: docs/2-advanced/07-accessibility.md [hunk 3]
```diff
@@ -55,18 +55,19 @@ For more information regarding the available themes and how to use them, see the
...
+| `accessibleRole`                            | `role`          | Sets the accessible aria role of the component.                                                                                                                                                     |
+| `accessibilityTexts` (FlexibleColumnLayout) | `aria-label`    | An object of strings that define several additional accessibility texts for even further customization.                                                                                             |
```
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:63]:
```suggestion
| `accessibilityTexts` (FlexibleColumnLayout) | `aria-label`    | An object of strings that define several additional accessibility texts for further customization.                                                                                             |
```

File: docs/2-advanced/07-accessibility.md [hunk 4]
```diff
@@ -55,18 +55,19 @@ For more information regarding the available themes and how to use them, see the
...
+| `accessibilityRoles` (FlexibleColumnLayout) | `role`          | An object of strings that define several additional accessibility roles for even further customization.                                                                                             |
```
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:64]:
```suggestion
| `accessibilityRoles` (FlexibleColumnLayout) | `role`          | An object of strings that define several additional accessibility roles for further customization.                                                                                             |
```

File: docs/2-advanced/07-accessibility.md (hunk 3 above)
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:63]:
```suggestion
| `accessibilityTexts` (`FlexibleColumnLayout`) | `aria-label`    | An object of strings that define several additional accessibility texts for even further customization.                                                                                             |
```

File: docs/2-advanced/07-accessibility.md (hunk 4 above)
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:64]:
```suggestion
| `accessibilityRoles` (`FlexibleColumnLayout`) | `role`          | An object of strings that define several additional accessibility roles for even further customization.                                                                                             |
```

File: docs/2-advanced/07-accessibility.md [hunk 5]
```diff
@@ -144,6 +145,31 @@ Will result in the shadow DOM as:
 
 ---
 
+### accessibilityRoles
+
+This property accepts `object` with properties values for different parts of the FlexibleColumnLayout elements. For more detailed information on every object property, read the API description in [FlexibleColumnLayout](https://sap.github.io/ui5-webcomponents/playground/components/FlexibleColumnLayout).
```
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:150]:
```suggestion
This property accepts `object` with property values for different parts of the `FlexibleColumnLayout` elements. For more detailed information on every object property, read the API description in [FlexibleColumnLayout](https://sap.github.io/ui5-webcomponents/playground/components/FlexibleColumnLayout). 
```

File: docs/2-advanced/07-accessibility.md [hunk 6]
```diff
@@ -144,6 +145,31 @@ Will result in the shadow DOM as:
...
+
+<script>
+    const component = document.getElemetnById("component");
+    component.accessibilityRoles = {
+        startColumnRole: "complimentary",
+    };
+</script>
+```
+
+Will result in the shadow DOM as:
```
Comment (xtatica) [inline docs/2-advanced/07-accessibility.md:164]:
```suggestion
Results in the shadow DOM as: 
```

File: packages/fiori/src/FlexibleColumnLayout.js [hunk 7]
```diff
@@ -103,6 +105,24 @@ const metadata = {
 			type: Object,
 		},
 
+		/**
+		 * An object of strings that defines several additional accessibility roles for even further customization.
```
Comment (xtatica) [inline packages/fiori/src/FlexibleColumnLayout.js:109]:
```suggestion
		 * An object of strings that defines additional accessibility roles for further customization.
```

**Review by xtatica — CHANGES_REQUESTED** [review]

**Review by nnaydenow — APPROVED** [review]

**Review by xtatica — APPROVED** [review]

**Pull request merged**